src/renderer/services/file-management-system/index.ts

```diff
@@ -398,85 +397,134 @@ export default class FileManagementSystem {
+    if (lastChunkNumber > 0) {
+      const chunkResponse = await this.fss.getChunkInfo(fssStatus.uploadId, lastChunkNumber);
+      partiallyCalculatedMd5 = chunkResponse.cumulativeMD5;
+      if (!partiallyCalculatedMd5) {
+        throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
+      }
+    }
+    return [lastChunkNumber, partiallyCalculatedMd5];
+  }
 
-        if (fssStatus.status === UploadStatus.RETRY) {
-          if (!partiallyCalculatedMd5){
-            throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
-          }
-          const deserailizedMd5Hasher = await Md5Hasher.deserialize(partiallyCalculatedMd5);
-          await this.fss.finalize(fssUploadId, deserailizedMd5Hasher.digest());
-        } else if (fssStatus.status === UploadStatus.WORKING) {
-          const { originalPath } = upload.serviceFields.files[0].file;
-          const { size: fileSize } = await fs.promises.stat(originalPath);
-          await this.uploadInChunks({
-            fssUploadId,
-            source: originalPath,
-            chunkSize: fssStatus.chunkSize,
-            user: upload.user,
-            onProgress: (bytesUploaded) =>
-              onProgress(upload.jobId, { bytesUploaded, totalBytes: fileSize }),
-            initialChunkNumber: lastChunkNumber,
-            partiallyCalculatedMd5
-          });
-        }
+  private async retryChunkedUpload(
+    fssStatus: UploadStatusResponse
+  ) {
+    const [lastChunkNumber, partiallyCalculatedMd5] = await this.getChunkedUploadProgress(fssStatus);
+    if (!partiallyCalculatedMd5) {
+      throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
+    }
+    const deserailizedMd5Hasher = await Md5Hasher.deserialize(partiallyCalculatedMd5);
+    await this.fss.retryFinalizeMd5(fssStatus.uploadId, deserailizedMd5Hasher.digest());
+  }
+
+  private async resumeUploadInChunks(
+    upload: UploadJob,
+    fssStatus: UploadStatusResponse,
+    onProgress: (uploadId: string, progress: UploadProgressInfo) => void
+  ) {
+    const { originalPath } = upload.serviceFields.files[0].file;
+    const { size: fileSize } = await fs.promises.stat(originalPath);
+    const [lastChunkNumber, partiallyCalculatedMd5] = await this.getChunkedUploadProgress(fssStatus);
+
+    await this.uploadInChunks({
+      fssStatus,
+      source: originalPath,
+      user: upload.user,
+      onProgress: (bytesUploaded) =>
+        onProgress(upload.jobId, { bytesUploaded, totalBytes: fileSize }),
+      initialChunkNumber: lastChunkNumber,
+      partiallyCalculatedMd5
+    });
+  }
+
+  private async updateProgress(
+    fssUploadId: string,
+    fileSize: number,
+    onProgress: (progress: UploadProgressInfo) => void) {
+    const fssStatusResponse = await this.fss.getStatus(fssUploadId);
+    let fssStatus = fssStatusResponse?.status;
+    while (fssStatus !== UploadStatus.COMPLETE) {
+      await FileManagementSystem.sleep(1000); //TODO too short?  Tests timeout if > 2 sec
```

Comment:
I personally prefer tracking TODOs like this that are easily resolvable outside of the codebase or at least outside the PR.

Reply:
I like TODOs because stopping to create a JIRA ticket breaks me out of the flow.  I go back and create the tickets later when i make a pr.  But I like to keep to TODOs because they are a placeholder and create a sense of pressure to go back and actually do it, rather than let it be forgotten in the backlog.